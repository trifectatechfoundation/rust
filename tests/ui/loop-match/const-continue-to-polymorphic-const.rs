// The value of a trait's associated constant is not known when lowering #[const_continue], and
// hence we can't determine which branch it should jump to.
#![allow(incomplete_features)]
#![feature(loop_match)]
#![crate_type = "lib"]

trait Foo {
    const Target: u8;

    fn test_u8(mut state: u8) -> &'static str {
        #[loop_match]
        loop {
            state = 'blk: {
                match state {
                    0 => {
                        #[const_continue]
                        break 'blk Self::Target;
                        //~^ ERROR could not determine the target branch for this `#[const_continue]`
                    }

                    1 => return "bar",
                    2 => return "baz",
                    _ => unreachable!(),
                }
            }
        }
    }
}
