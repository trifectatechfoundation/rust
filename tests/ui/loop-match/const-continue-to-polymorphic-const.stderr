error: could not determine the target branch for this `#[const_continue]`
  --> $DIR/const-continue-to-polymorphic-const.rs:17:36
   |
LL |                         break 'blk Self::Target;
   |                                    ^^^^^^^^^^^^ this value is too generic

error: aborting due to 1 previous error

